language_detector/main.py

```diff
@@ -1,9 +1,35 @@
 """This is the entry point of the program."""
+# coding=utf-8
 
 from .languages import LANGUAGES
 
-
 def detect_language(text, languages=LANGUAGES):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    text = text.strip().split("\n\n")
+    count = {}
+    
+    for t in text:
+        count[find_lang(t, languages=LANGUAGES)] = len(t)
+        
+    return max(count, key=count.get)
+
+
+def find_lang(text, languages=LANGUAGES):
+    for ch in "-(),.":
+        text = text.replace(ch, "")
+        
+    text = text.strip().split(" ")
+    text = [el.strip() for el in text if el != ""]
+
+    wordcount = {}
+    for word in text:
+        if word not in wordcount:
+            wordcount[word] = 1
+        else:
+            wordcount[word] += 1
+        
+    freq_word = max(wordcount, key=wordcount.get)
+
+    for language in languages:
+        if freq_word in language["common_words"]:
+            return language["name"]
```

Comment:
As I understand it, you basicly check which word in the text occurs the most in the text and look in which language dictionary it is. This might not always work, unless you have a realy extensive dictionary for every language, and there might be words that belong to two or more languages. So this might fail in some edge cases. 
If the most common word is not found in any dictionary the function will return None and your program should get an error.
Maybe try to map the given list of languages to int variables. Then count how many words of the text are in each language-dictionary and increase the variables accordingly.